### DISCUSSION TITLE
Why is the timeseries information generated by pyswmm (such as node depth timeseries data) different from that in swmm?

### DISCUSSION THREAD

**Discussion by jiayison** [pyswmm/pyswmm#539] [Q&A] [answered]
** WAIT!!! WAIT!!! Is this an ISSUE or DISCUSSION??**
If you need help with a problem, ask a question in our discussion board! if you are getting bad/wrong data or you are experiencing a bug due to pyswmm, raise an issue.   

**Describe the bug**
I find that there is something wrong with the timeseries data in the output module of pyswmm:
      1.I use the NodeSeries of pyswmm to get the invert_depth of all my nodes. But the results are different from the results in SWMM, and the difference is non-negligible.  
      2. Some nodes are exist in inp, but can't be found in the NodeSeries module of pyswmm.
 Here is the code:
 
pyswmm import Simulation, SimulationPreConfig, Output, NodeSeries
import pandas as pd
import csv
import os

folder_path = 'D:\\study\\experiment\\gaoxin\\data\\rainfall'

files = [f for f in os.listdir(folder_path) if f.endswith('.dat')]

for file in files:
    file_path = os.path.join(folder_path, file)

    sim_conf = SimulationPreConfig()
    sim_conf.add_update_by_token("TIMESERIES", "file", 2, file_path)

    with Simulation("swmm/gx_st_river_0611.inp", sim_preconfig=sim_conf) as sim:
        sim.execute()

    with Output('swmm/gx_st_river_0611_mod.out') as out:
        nodes_file = 'data/nodeID.csv'
        df_nodes = pd.read_csv(nodes_file, header=None, names=['nodename'])
        node_names = df_nodes['nodename'].tolist()

        csv_filename = f'data/node_depth/node_depth_{os.path.splitext(file)[0]}.csv'  # 每个文件生成一个唯一的CSV文件名
        with open(csv_filename, 'w', newline='') as csvfile:
            writer = csv.writer(csvfile)
            writer.writerow(['Time'] + node_names)

            time_steps = set()
            for node_name in node_names:
                if node_name in out.links:
                    ts = NodeSeries(out)[node_name].invert_depth
                    time_steps.update(ts.keys())

            sorted_time_steps = sorted(time_steps)

            for time_step in sorted_time_steps:
                data_row = [time_step]
                for node_name in node_names:
                    if node_name in out.links:
                        ts = NodeSeries(out)[node_name].invert_depth
                        data_row.append(ts[time_step] if time_step in ts else None)
                    else:
                        data_row.append('link does not exist in model output')
                writer.writerow(data_row)

**Comment by bemcdonnell** [comment]
@jiayison, can you share your before / after INP files? I’m wondering what the [TIMESERIES] sections look like.  Can you compare those to make sure the changes you put using SimulationPreConfig were applied correctly?

**Comment by jiayison** [comment]
> @jiayison, can you share your before / after INP files? I’m wondering what the [TIMESERIES] sections look like. Can you compare those to make sure the changes you put using SimulationPreConfig were applied correctly?

sorry sir, the INP filescan't be shared. But i just compared two files to find that the only difference is the file path of the TIMESERIES of rainfall, which is what i need. So Is there any possibility that something else might have gone wrong?

**Comment by bemcdonnell** [comment, marked as answer]
Just so I understand, you’re comparing EPA-SWMM run to pyswmm run? Also, what versions are you running of each?  Look at the header inside the RPT files that you’re comparing. Can you generate a time series for one location for both epaswmm and pyswmm? Screenshot the one from EPA SWMM and simplify your code a bit to just use NodeSeries for a single location.

**Comment by jiayison** [comment]
> Just so I understand, you’re comparing EPA-SWMM run to pyswmm run? Also, what versions are you running of each? Look at the header inside the RPT files that you’re comparing. Can you generate a time series for one location for both epaswmm and pyswmm? Screenshot the one from EPA SWMM and simplify your code a bit to just use NodeSeries for a single location.

Thank u for ur patience! I just test it with one node, and the retult is right. I think it is not the problem of pyswmm, so I will check other codes. Thanks again!
